src/language.cc

```diff
@@ -26,57 +26,60 @@ struct Db
-                                              { "eo_UY", QT_TR_NOOP( "Esperanto" ) },
-                                              { "de_CH", QT_TR_NOOP( "German, Switzerland" ) },
-                                              { "es_BO", QT_TR_NOOP( "Spanish, Bolivia" ) },
-                                              { "tg_TJ", QT_TR_NOOP( "Tajik" ) },
-                                              { "qu_PE", QT_TR_NOOP( "Quechua" ) },
-                                              { "ay_BO", QT_TR_NOOP( "Aymara" ) },
-                                              { "ar_SA", QT_TR_NOOP( "Arabic, Saudi Arabia" ) },
-                                              { "tk_TM", QT_TR_NOOP( "Turkmen" ) },
-                                              { "ie_001", QT_TR_NOOP( "Interlingue" ) },
-                                              { "jbo_EN", QT_TR_NOOP( "Lojban" ) },
-                                              { "hu_HU", QT_TR_NOOP( "Hungarian" ) },
-                                              { "en_US", QT_TR_NOOP( "English" ) } };
+  // Note: Keep this list sorted!
+  QMap< QString, QString > translationLangs = {
+    { QT_TR_NOOP( "Albanian" ), "sq" },
+    { QT_TR_NOOP( "Arabic" ), "ar" },
+    { QT_TR_NOOP( "Aymara" ), "ay" },
+    { QT_TR_NOOP( "Belarusian" ), "be" },
+    { QT_TR_NOOP( "Bulgarian" ), "bg" },
+    { QT_TR_NOOP( "Chinese Simplified" ), "zh_CN" },
+    { QT_TR_NOOP( "Chinese Traditional" ), "zh_TW" },
+    { QT_TR_NOOP( "Czech" ), "cs" },
+    { QT_TR_NOOP( "Dutch" ), "nl" },
+    { QT_TR_NOOP( "English" ), "en" },
+    { QT_TR_NOOP( "Esperanto" ), "eo" },
+    { QT_TR_NOOP( "Finnish" ), "fi" },
+    { QT_TR_NOOP( "French" ), "fr_FR" },
+    { QT_TR_NOOP( "German" ), "de" },
+    { QT_TR_NOOP( "German, Switzerland" ), "de_CH" },
+    { QT_TR_NOOP( "Greek" ), "el" },
+    { QT_TR_NOOP( "Hindi" ), "hi" },
+    { QT_TR_NOOP( "Hungarian" ), "hu" },
+    { QT_TR_NOOP( "Interlingue" ), "ie_001" },
+    { QT_TR_NOOP( "Italian" ), "it" },
+    { QT_TR_NOOP( "Japanese" ), "ja" },
+    { QT_TR_NOOP( "Korean" ), "ko" },
+    { QT_TR_NOOP( "Lithuanian" ), "lt" },
+    { QT_TR_NOOP( "Lojban" ), "jbo" },
+    { QT_TR_NOOP( "Macedonian" ), "mk" },
+    { QT_TR_NOOP( "Persian" ), "fa" },
+    { QT_TR_NOOP( "Polish" ), "pl" },
+    { QT_TR_NOOP( "Portuguese" ), "pt" },
+    { QT_TR_NOOP( "Portuguese, Brazilian" ), "pt_BR" },
+    { QT_TR_NOOP( "Quechua" ), "qu" },
+    { QT_TR_NOOP( "Russian" ), "ru" },
+    { QT_TR_NOOP( "Serbian" ), "sr" },
+    { QT_TR_NOOP( "Slovak" ), "sk" },
+    { QT_TR_NOOP( "Spanish" ), "es" },
+    { QT_TR_NOOP( "Spanish, Argentina" ), "es_AR" },
+    { QT_TR_NOOP( "Spanish, Bolivia" ), "es_BO" },
+    { QT_TR_NOOP( "Swedish" ), "sv" },
+    { QT_TR_NOOP( "Tajik" ), "tg" },
+    { QT_TR_NOOP( "Turkish" ), "tr" },
+    { QT_TR_NOOP( "Turkmen" ), "tk" },
+    { QT_TR_NOOP( "Ukrainian" ), "uk" },
+    { QT_TR_NOOP( "Vietnamese" ), "vi" },
+  };
```

Comment:
the key and value are swapped intendly?

Reply:
Yes.

We don't need a locale->language mapping.

We need a user-facing lang name -> translation lang code mapping.

Reply:
This is micro-optimization. It removed some unnecessary lookups in the for loops.